code/mob/living/critter/ai/flock/flocktasks.dm

```diff
@@ -713,9 +714,20 @@ butcher
 	var/run_range = 3
 	var/list/dummy_params = list("icon-x" = 16, "icon-y" = 16)
 
+/datum/aiTask/timed/targeted/flockdrone_shoot/proc/precondition()
+	//if we know there are enemies, and gun is charged and ready
+	var/mob/living/critter/flock/drone/F = holder.owner
+	if(length(F.flock?.enemies))
+		var/datum/handHolder/HH = F.hands[3]
+		var/datum/limb/gun/stunner = HH?.limb
+		if(istype(stunner) && !stunner.is_on_cooldown())
```

Comment:
This will mean a drone with its stunner on cooldown will just calmly revert to harvesting or something, rather than reacting like its in combat. Maybe a better change would be to make the drone dodge if their stunner is on cooldown during a shoot task, so they still react as if they're in combat and try to dodge until they can shoot back.

Reply:
My reasoning for this was that it's totally possible to essentially "combat lock" a bunch of drones by having them attack you but not be able to get you (see: an enemy pod parked outside a locked airlock or something). I figured if we make them do other stuff between gun charges, it's less likely to be a cheesey strategy because they'll be able to convert doors/walls and do other stuff between taking pot shots.